LDAP code does not build anymore

Currently I'm not able to build the LDAP code in Debian 13.

Running `cd build; OS=Linux LDAPINC= make -f Makefile.test-ldap test-compile` fails with the following errors:
```
test-ldap.c: In function ‘main’:
test-ldap.c:31:96: error: passing argument 2 of ‘ldap_url_parse’ from incompatible pointer type [-Wincompatible-pointer-types]
   31 |                 rc = ldap_url_parse("ldap://ldap.example.com/cn=xymon,ou=development,o=xymon", &ludp);
      |                                                                                                ^~~~~
      |                                                                                                |
      |                                                                                                LDAPURLDesc * {aka struct ldap_url_desc *}
In file included from /usr/include/lber_types.h:24,
                 from /usr/include/lber.h:29,
                 from /usr/include/ldap.h:30,
                 from test-ldap.c:5:
/usr/include/ldap.h:2079:16: note: expected ‘LDAPURLDesc **’ {aka ‘struct ldap_url_desc **’} but argument is of type ‘LDAPURLDesc *’ {aka ‘struct ldap_url_desc *’}
 2079 | ldap_url_parse LDAP_P((
      |                ^~~~~~
test-ldap.c:32:22: error: implicit declaration of function ‘ldap_init’ [-Wimplicit-function-declaration]
   32 |                 ld = ldap_init(ludp.lud_host, ludp.lud_port);
      |                      ^~~~~~~~~
test-ldap.c:32:20: error: assignment to ‘LDAP *’ {aka ‘struct ldap *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   32 |                 ld = ldap_init(ludp.lud_host, ludp.lud_port);
      |                    ^
make: *** [Makefile.test-ldap:4: test-compile] Fehler 1
```

On Ubuntu 24.04 the errors are only warnings, so this succeeds.

In Debian we have a patch the uses `#define LDAP_DEPRECATED 1` that works around the implicit declaration error, but this doesn't repair the incompatible pointer type problem (https://salsa.debian.org/debian/xymon/-/blob/master/debian/patches/33_526176-ldap.patch).

Cherry-picking the build/test-ldap.c part from https://github.com/xymon-monitoring/xymon/issues/22 seems to solve the issue.

--- a/build/test-ldap.c
+++ b/build/test-ldap.c
@@ -2,6 +2,7 @@
 #include <sys/time.h>
 #include <stdio.h>
 
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 #include <lber.h>
 
@@ -21,14 +22,14 @@ int main(int argc, char *argv[])
 		}
 	}
 	else {
-		LDAPURLDesc ludp;
+		LDAPURLDesc ludp,*pludp = &ludp;
 		int protocol, rc;
 		struct timeval nettimeout;
 
 		protocol = LDAP_VERSION3;
 		nettimeout.tv_sec = 1; nettimeout.tv_usec = 0;
 
-		rc = ldap_url_parse("ldap://ldap.example.com/cn=xymon,ou=development,o=xymon", &ludp);
+		rc = ldap_url_parse("ldap://ldap.example.com/cn=xymon,ou=development,o=xymon", &pludp);
 		ld = ldap_init(ludp.lud_host, ludp.lud_port);
 		rc = ldap_set_option(ld, LDAP_OPT_PROTOCOL_VERSION, &protocol);
 		rc = ldap_set_option(ld, LDAP_OPT_NETWORK_TIMEOUT, &nettimeout);

